benchmarks/datasets.py

```diff
@@ -83,17 +81,31 @@ def _synth_classification_dataset(n_samples=1000, n_features=10000,
     X, y = make_classification(n_samples=n_samples, n_features=n_features,
                                n_classes=n_classes, random_state=42,
                                n_informative=n_features, n_redundant=0)
-
     X = X.astype(dtype, copy=False)
-
     if representation is 'sparse':
         X[X < 2] = 0
         X = sp.csr_matrix(X)
-
     X = MaxAbsScaler().fit_transform(X)
     return X, y
 
 
+@M.cache
+def _decomposition_dataset():
+    """
```

Comment:
I'd name it `olivetti_faces_dataset` instead. It might also be used for non decomposition benchmarks later.